pandas/core/case_when.py

```diff
@@ -0,0 +1,125 @@
+from __future__ import annotations
+
+from typing import Any
+
+from pandas.core.dtypes.common import is_array_like
+
+import pandas as pd
+import pandas.core.common as com
+
+
+def case_when(obj: pd.DataFrame | pd.Series, *args, default: Any) -> pd.Series:
+    """
+    Returns a Series based on multiple conditions assignment.
+
+    This is useful when you want to assign a column based on multiple conditions.
+    Uses `Series.mask` to perform the assignment.
+
+    Parameters
+    ----------
+    obj : Dataframe or Series on which the conditions will be applied.
+    args : Variable argument of conditions and expected values.
+        Takes the form:
+            `condition0`, `value0`, `condition1`, `value1`, ...
+        `condition` can be a 1-D boolean array/series or a callable
+        that evaluate to a 1-D boolean array/series. See examples below.
```

Comment:
value0 et al can be Series here, yes?

Reply:
Correct. I added an example in the docstring: 

https://github.com/ELHoussineT/pandas/blob/3e6dc997d7b0a9f0c4b20763306f6003fcc559b1/pandas/core/case_when.py#L62-L71